mlir/include/Quantum/IR/QuantumOps.td

```diff
@@ -305,8 +333,8 @@ def SetBasisStateOp : Quantum_Op<"set_basis_state", []> {
     }];
 }
 
-def CustomOp : Gate_Op<"custom", [DifferentiableGate,
-                                  AttrSizedOperandSegments, AttrSizedResultSegments ]> {
+def CustomOp : Gate_Op<"custom", [DifferentiableGate, NoMemoryEffect,
```

Comment:
Why the NoMemoryEffect here?

Reply:
Let me know if the explanation above is sufficient :)